platform/CMakeLists.txt

```diff
@@ -0,0 +1,124 @@
+include_directories(${CMAKE_HOME_DIRECTORY}/3party/jansson/src ${PROJECT_SOURCE_DIR}/platform)
+
+set(
+    SRC
+    chunks_download_strategy.hpp
+    chunks_download_strategy.cpp
+    constants.hpp
+    country_defines.hpp
+    country_defines.cpp
+    country_file.hpp
+    country_file.cpp
+    file_logging.hpp
+    file_logging.cpp
+    get_text_by_id.hpp
+    get_text_by_id.cpp
+    http_request.hpp
+    http_request.cpp
+    http_thread_callback.hpp
+    local_country_file.hpp
+    local_country_file.cpp
+    local_country_file_utils.hpp
+    local_country_file_utils.cpp
+    location.hpp
+    measurement_utils.hpp
+    measurement_utils.cpp
+    mwm_traits.hpp
+    mwm_traits.cpp
+    mwm_version.hpp
+    mwm_version.cpp
+    platform.hpp
+    platform.cpp
+    preferred_languages.hpp
+    preferred_languages.cpp
+    servers_list.hpp
+    servers_list.cpp
+    settings.hpp
+    settings.cpp
+)
+
+if (NOT ${PL_IPHONE} AND NOT ${PL_ANDROID})
+    set(
+        SRC
+	${SRC}
+	platform_qt.cpp
+        wifi_location_service.cpp
+        location_service.cpp
+
+        wifi_info.hpp
+        location_service.hpp
+    )
+
+    if (NOT ${PL_MAC})
+        set(
+	    SRC
+	    ${SRC}
+            http_thread_qt.cpp
+            http_thread_qt.hpp
+        )
+    endif()
+
+
+    if (${PL_WIN})
+        set(
+	    SRC
+	    ${SRC}
+	    platform_win.cpp 
+            wifi_info_windows.cpp
+	)
+    elseif(${PL_MAC})
+        set(
+            SRC
+            ${SRC}
+            platform_mac.mm
+            apple_location_service.mm
+        )
+    elseif(${PL_LINUX})
+        set(
+	    SRC
+	    ${SRC}
+            platform_linux.cpp
+	)
+    endif()
+   
+elseif(${PL_IPHONE})
+    set(
+        SRC
+        ${SRC}
+        platform_ios.mm
+    )
+elseif(${PL_ANDROID})
+    set(
+        SRC
+	${SRC}
+	platform_android.cpp
+    )
+endif()
+
+if (${PL_MAC} OR ${PL_IPHONE})
```

Comment:
I can't merge it just by adding "else", i will need to do "elseif". I do not see any benefit in it.